dev/src/dev.clj

```diff
@@ -58,10 +59,28 @@
   (mbc/init!)
   (reset! initialized? true))
 
+(defn- deleted-inmem-databases
+  []
+  (let [h2-dbs (t2/select :model/Database :engine :h2)
+        in-memory? (fn [db] (some-> db :details :db (str/starts-with? "mem:")))
+        can-connect? (fn [db]
+                       (binding [metabase.driver.h2/*allow-testing-h2-connections* true]
+                         (try
+                           (driver/can-connect? :h2 (:details db))
+                           (catch org.h2.jdbc.JdbcSQLNonTransientConnectionException _
```

Comment:
I'm a bit hesitant to catch such a specific error here. This is in general just "annoying" and not super bad. I think we should catch all exceptions and perhaps not remove those dbs. I'm just thinking if a check throws an error besides this one, we won't continue startup and just end up hanging here.

Reply:
Good point, that makes sense.

Reply:
er, I think it's awesome you are catching that specific error and removing it. But we should catch others and not remove those. I realized I worded that a bit poorly

Reply:
Nah it made sense to me. Something like:

```clojure
(catch Exception e
  (log/error e "Error connecting to inmem database")
  ;; we don't want to delete these, so just pretend we could connect
  true)
```

along the lines of what you're thinking?

Reply:
```clojure
(try (driver/can-connect? :h2 db)
     (catch org.h2.jdbc.JdbcSQLNonTransientConnectionException _
         false)
     (catch Exception e
       (log/error "Error checking in-memory database for deletion" e)
       true))
```

something like this. I think it's valid for a db to throw an error if it cannot connect:

> This function should return truthy if a connection to the DB
  can be made successfully, otherwise it should return falsey or throw an appropriate Exception. Exceptions if a
  connection cannot be made. Throw an `ex-info` containing a truthy `::can-connect-message?` in `ex-data`
  in order to suppress logging expected driver validation messages during setup.

If this is throwing a NonTransient Connection Exception that sounds like there's no chance of it working in the future and it is validly gone. Other exceptions, who knows. So we'll leave those and we can manually clean them up.